core/clingutils/src/TClingUtils.cxx

```diff
@@ -4184,6 +4184,10 @@ void ROOT::TMetaUtils::GetNormalizedName(std::string &norm_name, const clang::Qu
    cling::Interpreter::PushTransactionRAII clingRAII(const_cast<cling::Interpreter*>(&interpreter));
    normalizedType.getAsStringInternal(normalizedNameStep1,policy);
 
+   norm_name = normalizedNameStep1;
+   TClassEdit::AtomicTypeNameHandlerRAII atomicTypeNameHandler_step1(normalizedNameStep1);
+   TClassEdit::AtomicTypeNameHandlerRAII atomicTypeNameHandler_norm_name(norm_name);
```

Comment:
The first line is a fully copy of the (possibly long type).  We might want to extend AtomicTypeNameHandlerRAII so that we can write 'just':
```suggestion
   TClassEdit::AtomicTypeNameHandlerRAII atomicTypeNameHandler_step1(normalizedNameStep1);
   TClassEdit::AtomicTypeNameHandlerRAII atomicTypeNameHandler_norm_name(norm_name, atomicTypeNameHandler_step1.IsAtomic());
```
where the second one just record the reference and trust the requested state.

Reply:
Thanks. I do not fully understand why you mention a "copy". The RAAI stores a reference to the string (see https://github.com/root-project/root/pull/18674/files#diff-bc1b5fbce8a4116a51b5d348e8510df04cc97eac8a74e75889bf86e561c59212R160 ) but maybe you are referring to something else - sorry to be thick.

Reply:
line 4187 makes a copy of the intermediary string.

Reply:
but isn't the suggested change require a copy anyway?

Reply:
ah maybe you mean checking if norm_name uses "_Atomic" as a type specifier, and if yes go through the expensive path and if not do as we do now, right?